### PR TITLE
Paradrop Production

### PR DESCRIPTION
**Pull request by obrakmann**
This is a cleaned-up version of #10266, which I promised @bynnar18 I'd submit on his behalf.

The original PR already had two +1s.

I added a test case, as well as a commit that moves `ProductionFromMapEdge` from d2k to common, because I think it's now more generally useful, for minigames and whatnot.

### CODE REVIEW

File: OpenRA.Mods.Common/Traits/ProductionParadrop.cs [hunk 1]
```diff
@@ -0,0 +1,170 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2015 The OpenRA Developers (see AUTHORS)
```
Comment (GraionDilach) [inline OpenRA.Mods.Common/Traits/ProductionParadrop.cs]: Outdated.

File: OpenRA.Mods.Common/Traits/Parachutable.cs [hunk 2]
```diff
@@ -54,7 +55,7 @@ public Parachutable(ActorInitializer init, ParachutableInfo info)
 			positionable = self.TraitOrDefault<IPositionable>();
 		}
 
-		public void OnLanded()
+		public void OnLanded(Actor ignore)
```
Comment (GraionDilach) [inline OpenRA.Mods.Common/Traits/Parachutable.cs]: Maybe make it explicit?

File: OpenRA.Mods.Common/Traits/ProductionParadrop.cs (hunk 1 above)
Comment (obrakmann) [inline OpenRA.Mods.Common/Traits/ProductionParadrop.cs]: Well-spotted, thanks.

File: OpenRA.Mods.Common/Traits/Parachutable.cs (hunk 2 above)
Comment (obrakmann) [inline OpenRA.Mods.Common/Traits/Parachutable.cs]: Yup, done.

**Comment by Mailaender** [comment]
:white_check_mark:

File: OpenRA.Mods.Common/Traits/ProductionParadrop.cs [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+namespace OpenRA.Mods.Common.Traits
+{
+	[Desc("Deliver the unit in production via paradrop.")]
+	public class ProductionParadropInfo : ProductionInfo, Requires<ExitInfo>
+	{
+		[Desc("Cargo aircraft used. Must have Aircraft trait.")]
+		[ActorReference(typeof(AircraftInfo))] public readonly string ActorType = "badr";
+
+		[Desc("Sound to play when dropping the unit.")]
+		public readonly string ChuteSound = "chute1.aud";
```
Comment (Mailaender) [inline OpenRA.Mods.Common/Traits/ProductionParadrop.cs]: I guess not hard-coding these is better especially for common traits.

**Comment by abcdefg30** [comment]
Looks good to me. :+1: / :white_check_mark:
You can remove the testcase now.

**Comment by obrakmann** [comment]
Test case removed

**Comment by abcdefg30** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/8afffdddad8ed6810c700ab271aa94e0e6f08801...5817de267c1b59d059f971603000b36bdcbd62c4)

**Comment by Bynnar18** [comment]
I'm pleased this was able to make it in. I've been so busy lately I was disappointed I couldn't finish it.

**Pull request merged**